doc/learn/syntax-trees-with-typescript.md

```diff
@@ -0,0 +1,277 @@
+---
+group: guide
+title: Typing syntax trees with TypeScript
+description: Guide shows how to use types packages to work with syntax trees
+author: Christian Murphy
+authorGithub: ChristianMurphy
+tags:
+  - typescript
+  - unist
+  - mdast
+  - hast
+  - xast
+published: 2020-06-09
+modified: 2020-06-11
+---
+
+## Working with syntax trees in TypeScript
+
+This guide will introduce you to using unist and unified with TypeScript.
+
+### Contents
+
+*   [The Basics](#the-basics)
+*   [UNIST](#unist)
+*   [MDAST (Markdown)](#mdast-markdown)
+*   [HAST (HTML)](#hast-html)
+*   [XAST (XML)](#xast-xml)
+*   [Summary](#summary)
+*   [Next steps](#next-steps)
```

Comment:
Run `npm run format` again to regenerate the toc

Reply:
updated